// react-content/lib/PageBuilder.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = PageBuilder;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

if (typeof exports !== 'undefined') Object.defineProperty(exports, 'babelPluginFlowReactPropTypes_proptype_PageBuilderProps', {
  value: require('react').PropTypes.shape({
    data: require('react').PropTypes.arrayOf(require('react').PropTypes.any).isRequired
  })
});
function PageBuilder({ data }) {
  if (!data.length) {
    return null;
  } else if (data.length === 1) {
    return renderComponent(data[0]);
  }

  return _react2.default.createElement(
    'div',
    null,
    data.map(renderRootComponent)
  );
}

PageBuilder.propTypes = {
  data: require('react').PropTypes.arrayOf(require('react').PropTypes.any).isRequired
};
function renderRootComponent(data, key) {
  const props = _extends({}, data[1], {
    key
  });
  const newData = [data[0], props, ...data.slice(2)];

  return renderComponent(newData);
}

function renderComponent(data) {
  const component = getComponent(data[0]);
  const props = data[1];
  const children = data.slice(2);

  return _react2.default.createElement(component, props, ...children);
}

function getComponent(name) {
  return name;
}
